picoCTF-web/api/apps/v1/team.py

```diff
@@ -138,6 +138,9 @@ def post(self):
             raise PicoException('Teachers may not join teams!', 403)
         req = team_change_req.parse_args(strict=True)
 
+        if req['team_name'] == current_user['username']:
```

Comment:
both exceptions should really be unnecessary now that the self-team has a valid password to check... but I can see the value in raising something less confusing than "incorrect password" in this case

Reply:
Password is indeed redundant whatever the exception message, but didn't want to leave an invalid value in the password field.